polyply/tests/test_lib_files.py

```diff
@@ -0,0 +1,244 @@
+        return a1[1] == a2[1] and frozenset([a1[0], a1[2]]) == frozenset([a2[0], a2[2]]) and p1 == p2
+
+    return False
+
+@pytest.mark.parametrize("library, polymer", [
+     ['gromos2016H66', 'PP'],
+     ['gromos2016H66', 'C12E4'],
+     ['gromos2016H66', 'PE'],
+     ['gromos2016H66', 'PVA'],
+     ['gromos2016H66', 'PMA'],
+     ['gromos2016H66', 'PS'],
+   # ['martini3', 'PEO'],
+   # ['martini3', 'PS'],
+   # ['martini3', 'PEO_OHter'],
+   # ['martini3', 'P3HT'],
+    ['martini2', 'PEO'],
+    ['martini2', 'PS'],
+  # -> martini2 PEGylated lipids
+  # -> proteins?
+  # -> m2 polymers
+  #  ['gromos56A3', 'P3HT'],
+])
+def test_integration_protein(tmp_path, monkeypatch, library, polymer):
+    """
+    Runs tests on the library by executing the contents of the file `command` in
+    the folder libname/polymer, and tests whether the contents of the produced
+    files are the same as the reference files. The comparison of the files is
+    governed by `COMPARERS`.
+
+    Parameters
+    ----------
+    tmp_path
+    library: str
+    polymer: str
+    """
+    monkeypatch.chdir(tmp_path)
+
+    data_path = Path(PATTERN.format(path=INTEGRATION_DATA, library=library, polymer=polymer))
+
+    with open(str(data_path / 'command')) as cmd_file:
+        command = cmd_file.read().strip()
+
+    assert command  # Defensive
+    command = shlex.split(command)
+    result = [sys.executable]
+    for token in command:
+        if token.startswith('polyply'):  # Could be martinize2.py
+            result.append(str(POLYPLY))
+        elif token.startswith('.'):
+            result.append(str(data_path / token))
+        else:
+            result.append(token)
+
+    command = result
+
+    # read the citations that are expected
+    # citations = []
+    # with open(str(data_path/'citation')) as cite_file:
+    #     for line in cite_file:
+    #         citations.append(line.strip())
+
+    proc = subprocess.run(command, cwd='.', timeout=60, check=False,
+                          stdout=subprocess.PIPE,
+                          stderr=subprocess.PIPE,
+                          universal_newlines=True)
+    exit_code = proc.returncode
+    if exit_code:
+        print(proc.stdout)
+        print(proc.stderr)
+        assert not exit_code
+
+    # check if strdout has citations in string
+    #for citation in citations:
+    #    assert citation in proc.stderr
+
+    files = list(tmp_path.iterdir())
+
+    assert files
+    assert list(tmp_path.glob('*.itp')), files
+
+    for new_file in tmp_path.iterdir():
+        filename = new_file.name
+        reference_file = data_path/filename
+        assert reference_file.is_file()
+        ext = new_file.suffix.lower()
+        if ext in COMPARERS:
+            with monkeypatch.context() as m:
+                # Compare Interactions such that rounding errors in the
+                # parameters are OK.
+                m.setattr(vermouth.molecule.Interaction, '__eq__', _interaction_equal)
```

Comment:
With how you set it up this should no longer be needed I think?

Reply:
I don't know. What does the monkeypatch thing actually do?

Reply:
It temporarily (in the context) switches `vermouth.molecule.Interactions.__eq__` to `_interaction_equal`. So whenever you do `interaction_a == interaction_b` in that context, instead of calling `Interaction.__eq__` it calls `_interaction_equal`.